packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1251,3 +1252,30 @@ function IsInScope(context: TCGCContext, entity: Operation): boolean {
   }
   return true;
 }
+
+export const $deserializeEmptyStringAsNull: DeserializeEmptyStringAsNullDecorator = (
+  context: DecoratorContext,
+  target: ModelProperty,
+  scope?: LanguageScopes,
+) => {
+  if (target.kind !== "ModelProperty" || target.type.kind !== "Scalar") {
+    reportDiagnostic(context.program, {
+      code: "invalid-deserializeEmptyStringAsNull-target-type",
+      format: {},
+      target: target,
+    });
+    return;
+  }
+
+  if (target.type.name !== "string") {
+    const scalarType = target.type as Scalar;
+    if (scalarType.baseScalar === undefined || scalarType.baseScalar.name !== "string") {
```

Comment:
this could not handle multi-layer extends for scalar, you need to find the root base scalar.
```
scalar armid extends uuid;
scalar uuid extends string;
```

Reply:
the logic of finding base Scalar type has been refined.